### PR TITLE
Moves from Heroku to Rancher (fixes #687)

### LINKED ISSUE #687: Move from Heroku to Rancher
- [x] Get docker compose running
- [x] Configure travis to push to Docker hub if master
- [x] Deploy rancher stack
- [x] Get HTTPS cert
- [ ] Flip over CNAME
- [x] File a ticket for psql backups (#690)
- [x] Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by rickr**
Adds docker bits
Adds a Makefile for common tasks
Adds blog build script
Adds Docker Hub publish script
Adds script to update rancher
Updates travis config
Updates CONTRIBUTING with new docker based instructions

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -95,74 +45,24 @@ So, you want to learn how to program? Contributing to Operation Code is a great
...
+  make setup
+  ```
 
-  #### Rails on Cloud9:
-  * Cloud9 is a web-based integrated development environment. If you do not much experience in setting up a work environment, we highly recommend Cloud9 to start.
-  * [Getting Rails to work in Cloud9 in 5 easy steps!](https://ashtemp.github.io/Rails.html)
-  * When cloning your fork into Cloud9, create your workspace name and description, then select Ruby as the template.
-  * In the terminal, type in the following commands:
+  #### Running Operation Code:
+  Operation Code has some handy shorcuts built in to make common tasks simple. You can check out the [Makefile](https://github.com/operationcode/operationcode/blob/master/Makefile) to see a full listing of what these shortcuts are and what they do.
```
Comment (hollomancer) [inline CONTRIBUTING.md]: change to 'shortcuts'

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -95,74 +45,24 @@ So, you want to learn how to program? Contributing to Operation Code is a great
...
 
+  To run Operation Code you can use one of the following:
```
Comment (hollomancer) [inline CONTRIBUTING.md]: To run Operation Code (omit "use one of the following")

File: CONTRIBUTING.md (hunk 1 above)
Comment (rickr) [inline CONTRIBUTING.md]: e103e42

File: CONTRIBUTING.md (hunk 2 above)
Comment (rickr) [inline CONTRIBUTING.md]: e103e42

**Comment by rickr** [comment]
redis will be running in its own container now so we shouldn't see any more rdbs.

**Pull request merged**